etc/taskcluster/macos/states/generic-worker.sls

```diff
@@ -16,21 +17,76 @@
     - mode: 755
     - makedirs: True
 
-/etc/generic-worker/config.json:
+{{ user }}:
+  user.present:
+    - home: {{ home }}
+
+# `user.present`’s `createhome` is apparently not supported on macOS
+{{ home }}:
+  file.directory:
+    - user: {{ user }}
+
+{{ home }}/config.json:
   file.serialize:
     - makedirs: True
+    - user: {{ user }}
     - mode: 600
     - show_changes: False
     - formatter: json
     - dataset:
         provisionerId: proj-servo
         workerType: macos
-        workerId: servo-macos-1
-        clientId: project/servo/worker/macos/1
+        workerGroup: servo-macos
+        workerId: mac1
+        tasksDir: {{ home }}/tasks
         publicIP: {{ salt.network.ip_addrs()[0] }}
-        signingKeyLocation: {{ keyfile }}
+        signingKeyLocation: {{ home }}/key
+        clientId: {{ pillar["client_id"] }}
+        accessToken: {{ pillar["access_token"] }}
+        livelogSecret: {{ pillar["livelog_secret"] }}
 
-generic-worker new-openpgp-keypair --file {{ keyfile }}:
+{{ bin }}/generic-worker new-openpgp-keypair --file {{ home }}/key:
   cmd.run:
-    - creates: {{ keyfile }}
-    - prepend_path: {{ bin }}
+    - creates: {{ home }}/key
+    - runas: worker
+
+/Library/LaunchAgents/net.generic.worker.plist:
+  file.managed:
+    - mode: 644
+    - template: jinja
+    - contents: >-
```

Comment:
is it possible to store this file separately?

<s>i suppose the substs make this harder</s>

nah, we use files with substs in our current salt setup: https://github.com/servo/saltfs/blob/master/homu/init.sls#L69

Reply:
Done.